### ISSUE TITLE
addDoubletsScores  :  Error in H5Fopen(file, flags = flags, fapl = fapl, native = native) : HDF5. File accessibility. Unable to open file.

### ISSUE THREAD

**Issue by matteozoia4** [GreenleafLab/ArchR#1581]
**Attach your log file**
arrowfiles log :
[ArchR-createArrows-10a5f145ba289-Date-2022-08-22_Time-22-10-42.log](https://github.com/GreenleafLab/ArchR/files/9400444/ArchR-createArrows-10a5f145ba289-Date-2022-08-22_Time-22-10-42.log)

doubscore log : 
[ArchR-addDoubletScores-10a5f62d31d9f-Date-2022-08-23_Time-09-40-44.log](https://github.com/GreenleafLab/ArchR/files/9400451/ArchR-addDoubletScores-10a5f62d31d9f-Date-2022-08-23_Time-09-40-44.log)

**Describe the bug**
```
doubScores <- addDoubletScores(
     input = inputFiles,
     k = 10,
     knnMethod = "UMAP", 
     LSIMethod = 1)
```

ArchR logging to : ArchRLogs/ArchR-addDoubletScores-10a5f62d31d9f-Date-2022-08-23_Time-09-40-44.log
If there is an issue, please report to github with logFile!
2022-08-23 09:40:44 : Batch Execution w/ safelapply!, 0 mins elapsed.
Error in H5Fopen(file, flags = flags, fapl = fapl, native = native) : 
  HDF5. File accessibility. Unable to open file.>

**To Reproduce**
I so far didn't had time to test your provided files, I was trying to fix the bug by looking at previous discussion and issues on the topic (without positive outcome though). 

```
library(ArchR)
ArchR::installExtraPackages()

library(ArchR)
library(nloptr)
library(parallel)
library(stringi)
library(rhdf5)

library(BSgenome.Mmusculus.UCSC.mm10)
genomeAnnotation <- createGenomeAnnotation(genome = BSgenome.Mmusculus.UCSC.mm10)

library(TxDb.Mmusculus.UCSC.mm10.ensGene)
library(org.Mm.eg.db)
geneAnnotation <- createGeneAnnotation(TxDb = TxDb.Mmusculus.UCSC.mm10.ensGene, OrgDb = org.Mm.eg.db)
geneAnnotation

geneAnnotation <- createGeneAnnotation(
  TSS = geneAnnotation$TSS, 
  exons = geneAnnotation$exons, 
  genes = geneAnnotation$genes
)

inputFiles <- c(scATAC_E9.5="~/Documents/220325_PhD_MZ_back-up/Thesis_Project/In_progress/Multiom_Analysis/Files_for_ArchR/9.5_atac_fragments.tsv.gz", 
                scATAC_E10.5="~/Documents/220325_PhD_MZ_back-up/Thesis_Project/In_progress/Multiom_Analysis/Files_for_ArchR/10.5_atac_fragments.tsv.gz", 
                scATAC_E11.5="~/Documents/220325_PhD_MZ_back-up/Thesis_Project/In_progress/Multiom_Analysis/Files_for_ArchR/11.5_atac_fragments.tsv.gz")
inputFiles 

addArchRThreads(threads = 1)
addArchRGenome("mm10")

ArrowFiles <- createArrowFiles(
  inputFiles = inputFiles,
  sampleNames = names(inputFiles),
  minTSS = 4, 
  minFrags = 1000, 
  addTileMat = TRUE,
  addGeneScoreMat = TRUE
)

ArrowFiles

doubScores <- addDoubletScores(
  input = inputFiles,
  k = 10, 
  knnMethod = = "UMAP",
  LSIMethod = 1
  )>
```

**Expected behavior**
I would like to add the doubletscores to the 3 ATAC samples I produced the arrow files for.

**Screenshots**
(Not applicable)

**Session Info**
R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.0.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

Random number generation:
 RNG:     L'Ecuyer-CMRG 
 Normal:  Inversion 
 Sample:  Rejection 
 
locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Rsamtools_2.12.0                       org.Mm.eg.db_3.15.0                    TxDb.Mmusculus.UCSC.mm10.ensGene_3.4.0
 [4] GenomicFeatures_1.48.3                 AnnotationDbi_1.58.0                   BSgenome.Mmusculus.UCSC.mm10_1.4.3    
 [7] BSgenome_1.64.0                        rtracklayer_1.56.1                     Biostrings_2.64.1                     
[10] XVector_0.36.0                         stringi_1.7.8                          nloptr_2.0.3                          
[13] rhdf5_2.40.0                           SummarizedExperiment_1.26.1            Biobase_2.56.0                        
[16] MatrixGenerics_1.8.1                   Rcpp_1.0.9                             Matrix_1.4-1                          
[19] GenomicRanges_1.48.0                   GenomeInfoDb_1.32.3                    IRanges_2.30.1                        
[22] S4Vectors_0.34.0                       BiocGenerics_0.42.0                    matrixStats_0.62.0                    
[25] data.table_1.14.2                      stringr_1.4.1                          plyr_1.8.7                            
[28] magrittr_2.0.3                         ggplot2_3.3.6                          gtable_0.3.0                          
[31] gtools_3.9.3                           gridExtra_2.3                          ArchR_1.0.2                           

loaded via a namespace (and not attached):
  [1] utf8_1.2.2               reticulate_1.25          tidyselect_1.1.2         RSQLite_2.2.16           htmlwidgets_1.5.4       
  [6] BiocParallel_1.30.3      Rtsne_0.16               devtools_2.4.4           munsell_0.5.0            codetools_0.2-18        
 [11] ica_1.0-3                future_1.27.0            miniUI_0.1.1.1           withr_2.5.0              spatstat.random_2.2-0   
 [16] colorspace_2.0-3         progressr_0.10.1         filelock_1.0.2           rstudioapi_0.13          Seurat_4.1.1            
 [21] ROCR_1.0-11              tensor_1.5               listenv_0.8.0            labeling_0.4.2           GenomeInfoDbData_1.2.8  
 [26] harmony_0.1.0            polyclip_1.10-0          farver_2.1.1             bit64_4.0.5              pheatmap_1.0.12         
 [31] parallelly_1.32.1        vctrs_0.4.1              generics_0.1.3           BiocFileCache_2.4.0      R6_2.5.1                
 [36] doParallel_1.0.17        clue_0.3-61              bitops_1.0-7             rhdf5filters_1.8.0       spatstat.utils_2.3-1    
 [41] cachem_1.0.6             DelayedArray_0.22.0      assertthat_0.2.1         BiocIO_1.6.0             promises_1.2.0.1        
 [46] scales_1.2.1             rgeos_0.5-9              Cairo_1.6-0              globals_0.16.0           processx_3.7.0          
 [51] goftest_1.2-3            rlang_1.0.4              GlobalOptions_0.1.2      splines_4.2.1            lazyeval_0.2.2          
 [56] spatstat.geom_2.4-0      yaml_2.3.5               BiocManager_1.30.18      reshape2_1.4.4           abind_1.4-5             
 [61] httpuv_1.6.5             tools_4.2.1              usethis_2.1.6            ellipsis_0.3.2           spatstat.core_2.4-4     
 [66] RColorBrewer_1.1-3       sessioninfo_1.2.2        ggridges_0.5.3           progress_1.2.2           zlibbioc_1.42.0         
 [71] purrr_0.3.4              RCurl_1.98-1.8           ps_1.7.1                 prettyunits_1.1.1        rpart_4.1.16            
 [76] deldir_1.0-6             pbapply_1.5-0            GetoptLong_1.0.5         cowplot_1.1.1            urlchecker_1.0.1        
 [81] zoo_1.8-10               SeuratObject_4.1.0       ggrepel_0.9.1            cluster_2.1.3            fs_1.5.2                
 [86] scattermore_0.8          circlize_0.4.15          lmtest_0.9-40            RANN_2.6.1               fitdistrplus_1.1-8      
 [91] pkgload_1.3.0            hms_1.1.2                patchwork_1.1.2          mime_0.12                xtable_1.8-4            
 [96] XML_3.99-0.10            shape_1.4.6              biomaRt_2.52.0           compiler_4.2.1           tibble_3.1.8            
[101] KernSmooth_2.23-20       crayon_1.5.1             htmltools_0.5.3          mgcv_1.8-40              later_1.3.0             
[106] tidyr_1.2.0              DBI_1.1.3                dbplyr_2.2.1             ComplexHeatmap_2.12.1    rappdirs_0.3.3          
[111] MASS_7.3-58.1            cli_3.3.0                igraph_1.3.4             pkgconfig_2.0.3          GenomicAlignments_1.32.1
[116] sp_1.5-0                 plotly_4.10.0            spatstat.sparse_2.1-1    xml2_1.3.3               foreach_1.5.2           
[121] callr_3.7.2              digest_0.6.29            sctransform_0.3.4        RcppAnnoy_0.0.19         spatstat.data_2.2-0     
[126] leiden_0.4.2             uwot_0.1.14              curl_4.3.2               restfulr_0.0.15          shiny_1.7.2             
[131] rjson_0.2.21             lifecycle_1.0.1          nlme_3.1-159             jsonlite_1.8.0           Rhdf5lib_1.18.2         
[136] viridisLite_0.4.1        fansi_1.0.3              pillar_1.8.1             lattice_0.20-45          KEGGREST_1.36.3         
[141] fastmap_1.1.0            httr_1.4.4               pkgbuild_1.3.1           survival_3.4-0           glue_1.6.2              
[146] remotes_2.4.2            png_0.1-7                shinythemes_1.2.0        rhandsontable_0.3.8      iterators_1.0.14        
[151] bit_4.0.4                presto_1.0.0             profvis_0.3.7            blob_1.2.3               memoise_2.0.1           
[156] dplyr_1.0.9              irlba_2.3.5              future.apply_1.9.0   

**Additional context**
Nothing to add.

Many thanks for your help!
MZ

**Comment by rcorces** [comment]
Hi @matteozoia4! Thanks for using ArchR! Please make sure that your post belongs in the Issues section. Only bugs and error reports belong in the Issues section. Usage questions and feature requests should be posted in the [Discussions](https://github.com/GreenleafLab/ArchR/discussions) section, not in Issues.<br />
__Before we help you, you must respond to the following questions__ unless your original post already contained this information:
__1.__ If you've encountered an error, have you already searched previous Issues to make sure that this hasn't already been solved?
__2.__ Can you recapitulate your error using the tutorial code and dataset? If so, provide a reproducible example.
__3.__ Did you post your log file? If not, add it now.
__4.__ Remove any screenshots that contain text and instead copy and paste the text using markdown's codeblock syntax (three consecutive backticks). You can do this by editing your original post.

**Comment by matteozoia4** [comment]
Dear rcorces,

I managed to run the addDoubletsScores by re-setting the input files path (as shown below),  is that what I was missing as step?

```
input <- c(scATAC_E9.5="~/Documents/220325_PhD_MZ_back-up/Thesis_Project/In_progress/Multiom_Analysis/Files_for_ArchR/scATAC_E9.5.arrow", 
           scATAC_E10.5="~/Documents/220325_PhD_MZ_back-up/Thesis_Project/In_progress/Multiom_Analysis/Files_for_ArchR/scATAC_E10.5.arrow", 
           scATAC_E11.5="~/Documents/220325_PhD_MZ_back-up/Thesis_Project/In_progress/Multiom_Analysis/Files_for_ArchR/scATAC_E11.5.arrow")

doubScores <- addDoubletScores(
  input = input,
  k = 10, #Refers to how many cells near a "pseudo-doublet" to count.
  knnMethod = "UMAP", #Refers to the embedding to use for nearest neighbor search with doublet projection.
  LSIMethod = 1
  )

_ArchR logging to : ArchRLogs/ArchR-addDoubletScores-10a5f45314c2d-Date-2022-08-23_Time-11-49-56.log
If there is an issue, please report to github with logFile!
2022-08-23 11:49:57 : Batch Execution w/ safelapply!, 0 mins elapsed.
2022-08-23 11:49:57 : scATAC_E9.5 (1 of 3) :  Computing Doublet Statistics, 0 mins elapsed.
Filtering 1 dims correlated > 0.75 to log10(depth + 1)_
```

Many thanks for your time and patience!
MZ

**Comment by rcorces** [comment]
Please read the full manual and all relevant function documentation before posting. As illustrated in the manual and in the function documentation, the input to `addDoubletScores()` is not a vector of fragment files.

https://www.archrproject.com/bookdown/inferring-scatac-seq-doublets-with-archr.html
https://www.archrproject.com/reference/addDoubletScores.html

<html>
<body>


input | An ArchRProject object or a character vector containing the paths to the ArrowFiles to be used.
-- | --



</body>
</html>

**Comment by IqRaZaFaR16** [comment]
projHeme5 <- addArchRAnnotations(ArchRProj = projHeme5, collection = "EncodeTFBS")
ArchR logging to : ArchRLogs/ArchR-addArchRAnnotations-19a5c72d2a4560-Date-2024-12-05_Time-14-31-56.261448.log
If there is an issue, please report to github with logFile!
Error in H5Fopen(file, flags = flags, fapl = fapl, native = native) :
  HDF5. File accessibility. Unable to open file.
>as per mentioned ENCODE file cant open?
